source/fe/fe_values.cc

```diff
@@ -5009,6 +5074,32 @@ FESubfaceValues<dim, spacedim>::reinit(
 
 
 
+template <int dim, int spacedim>
+void
+FESubfaceValues<dim, spacedim>::reinit(
+  const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+  const TriaIterator<TriaAccessor<dim - 1, dim, spacedim>> &  face,
+  const TriaIterator<TriaAccessor<dim - 1, dim, spacedim>> &  subface)
+{
+  const auto faces   = cell->face_iterators();
+  const auto face_it = std::find(faces.begin(), faces.end(), face);
+  Assert(face_it != faces.end(),
```

Comment:
You are repeating these 3+ lines of code many times. Would it make sense to add a function that does this? Maybe ``cell->face_iterator_to_index(face)`` or ``face->my_child_index()`` or something? 

I am also not sure if we really need a linear search here, is there no identifying info inside the iterator we can use?

Reply:
Good idea: I should add that (I will repeat this search even more times when I write the `hp` version of this PR).

There is no such identifying information in the iterator since it could belong to two different cells.

Reply:
I added two such index functions in f791b78827fa0444093b31dcb2f48ba6e1a32d82.